code/datums/traits/good.dm

```diff
@@ -96,6 +96,33 @@
 	)
 	H.equip_in_one_of_slots(B, slots , qdel_on_fail = TRUE)
 
+/datum/quirk/multilingual
+	name = "Multilingual"
+	desc = "You spent a portion of your life learning to understand an additional language. You may or may not be able to speak it based on your anatomy."
+	value = 2
+	mob_trait = TRAIT_MULTILINGUAL
+	gain_text = "<span class='notice'>You have learned to understand an additional language.</span>"
+	lose_text = "<span class='danger'>You have forgotten how to understand a language.</span>"
+
+/datum/quirk/multilingual/on_spawn()
+	var/mob/living/carbon/human/H = quirk_holder
+	var/datum/language_holder/LH
+
+	var/datum/language/newLang
+	if(H.job != "Curator")
+		LH.add_blocked_language(/datum/language/narsie, LANGUAGE_ATOM)
+		LH.add_blocked_language(/datum/language/ratvar, LANGUAGE_ATOM)
+		LH.add_blocked_language(/datum/language/codespeak, LANGUAGE_ATOM)
+		LH.grant_all_languages(TRUE, TRUE, FALSE, LANGUAGE_ATOM)
+		do
+			newLang = LH.get_random_understood_language()
+		while(H.has_language(newLang, FALSE) == TRUE)
```

Comment:
Would be nice to have a sanity check just in case somehow, someone has all understood languages when they gain the quirk to stop the server getting stuck in a while loop and dying

Reply:
That loop should only execute if the player is not a Curator(the only setup where this would be applied, to my understanding) as it is applied under that initial if(). Unless AI initialization is weird and also runs quirks through it, this shouldn't happen.

Reply:
Iterative counter added. If somehow 15 random nonantag language rolls don't yield a language not possessed yet, the loop will break. No More Infinite Loops!